pkg/service/auth_context.go

```diff
@@ -27,78 +27,76 @@ type AuthContext struct {
 
 // AuthObjectConfig provides an interface for APIConfig objects that implements a Call method
 type AuthObjectConfig interface {
-	Call(ctx internal.AuthContext) (interface{}, error)
+	Call(ctx common.AuthContext) (interface{}, error)
 }
 
-func (authContext *AuthContext) getAuthObject(authObjConfig AuthObjectConfig) (interface{}, error) {
-	if ret, err := authObjConfig.Call(authContext); err != nil {
-		return nil, err
-	} else {
-		return ret, nil
-	}
-}
+type configCallback = func(config AuthObjectConfig, obj interface{})
 
-// GetIDObject gets an Identity auth object given an Identity config.
-func (authContext *AuthContext) GetIDObject() error {
+func (authContext *AuthContext) getAuthObject(configs []AuthObjectConfig, cb configCallback) error {
 	var wg sync.WaitGroup
 	var authObjError error
-	for _, config := range authContext.API.IdentityConfigs {
+	for _, config := range configs {
 		wg.Add(1)
-		var authObjCfg AuthObjectConfig = &config
 		go func() {
 			defer wg.Done()
-			if authObj, err := authContext.getAuthObject(authObjCfg); err != nil {
+			if authObj, err := config.Call(authContext); err != nil {
 				authObjError = err
 				fmt.Errorf("Invalid identity config", err)
 			} else {
-				authContext.Identity[&config] = authObj
+				cb(config, authObj)
 			}
 		}()
 	}
 	wg.Wait()
 	return authObjError
 }
 
+// GetIDObject gets an Identity auth object given an Identity config.
+func (authContext *AuthContext) GetIDObject() error {
+	configs := make([]AuthObjectConfig, len(authContext.API.IdentityConfigs))
+	// Convert []SpecificType to []interfaceType
+	for i, conf := range authContext.API.IdentityConfigs {
+		cpConf := conf
+		configs[i] = &cpConf
+	}
```

Comment:
This can't be like...

```suggestion
	for i, conf := range authContext.API.IdentityConfigs {
		configs[i] = &conf
	}
```

?

Reply:
No, you cannot do that. You need to copy the data: https://github.com/3scale-labs/authorino/pull/34#discussion_r560129840

Reply:
anyway, having a map with type pointers as keys has a bad smell that needs reviewing